ts/parts/StockChart.ts

```diff
@@ -1092,3 +1101,38 @@ addEvent(Chart, 'update', function (
         delete options.scrollbar;
     }
 });
+
+// Extend the Axis prototype to calculate start min/max values
+// (including min/maxPadding). This is related to using vertical panning
+// (#11315).
+addEvent(Axis, 'afterSetScale', function (
+    this: Highcharts.Axis
+): void {
+    var axis = this,
+        panning = axis.chart.options.chart &&
+            axis.chart.options.chart.panning,
+        options = axis.options as Highcharts.PanAxisOptions;
+
+    if (
+        panning &&
+        (panning.type === 'y' ||
+        panning.type === 'xy') &&
+        !axis.isXAxis &&
+        !defined(options.startMin) &&
+        !defined(options.startMax)
+    ) {
+
+        var min = Number.MAX_VALUE,
+            max = Number.MIN_VALUE;
+
+        axis.series.forEach(function (series): void {
+            min = Math.min(H.arrayMin(series.yData as any), min) -
+                (axis.min && axis.dataMin ? axis.dataMin - axis.min : 0);
+            max = Math.max(H.arrayMax(series.yData as any), max) +
+                (axis.max && axis.dataMax ? axis.max - axis.dataMax : 0);
+        });
+
+        options.startMin = min;
```

Comment:
We shouldn't write internal properties/state back to the `options`. Find another way place for it, either directly on the `Axis`, or, maybe better, create a preliminary `Axis.panning` or `Axis.panningState` object for it. Then the whole object can be destroyed when the panning finishes.

Reply:
After moving those values to separate object located directly on `Axis`, works fine.
Regarding your last sentence, I wondering whether understood it correctly, because I think it would be better to calculate those `start` values once, and keep them on the `Axis.panningState` object, than calculating/removing them on every panning action. Could you try to explain it in more detail?

Reply:
Well.. I don't have the full details, so I just suggested an architecture. If there are implementation problems with it, then of course that is more important.